addons/account/views/report_invoice.xml

```diff
@@ -210,6 +183,76 @@
             </t>
         </template>
 
+        <template id="document_tax_totals">
+            <!--
+                Generic template to display tax totals in pdf reports.
+                Used by invoices, SO and PO.
+
+                ARGUMENTS:
+                - tax_totals: dict in the form generated by account.move's _get_tax_totals.
+            -->
+            <!--Total amount without taxes-->
+            <tr class="border-black o_subtotal" style="">
+                <td><strong>Subtotal</strong></td>
+                <td class="text-right">
+                    <span t-esc="tax_totals['amount_untaxed']"/>
+                </td>
+            </tr>
+
+            <t t-set="multiple_tax_groups" t-value="len({group['tax_group_id'] for group_list in tax_totals['groups_by_total_type'].values() for group in group_list}) > 1"/>
+
+            <!-- Main total taxes-->
+            <t t-set="total_types_to_show" t-value="{'main_total'}"/>
+            <t t-call="account.tax_groups_totals"/>
+
+            <!--Total amount with taxes-->
+            <tr class="border-black o_total">
+                <td><strong>Total</strong></td>
+                <td class="text-right">
+                    <span t-esc="tax_totals['amount_total']"/>
+                </td>
+            </tr>
+
+            <!--After total taxes-->
+            <t t-set="total_types_to_show" t-value="{'after_total'}"/>
+            <t t-call="account.tax_groups_totals"/>
+        </template>
+
+        <template id="tax_groups_totals">
+            <!--
+                Generic template to display a list of tax groups with the related amounts.
+
+                ARGUMENTS:
+                - tax_totals: dict in the form generated by account.move's _get_tax_totals.
+                - total_types_to_show: the total_type keys to list the groups of
+                - multiple_tax_groups: true if more than one tax group is invovled; meaning
+                                       base amounts have to be displayed
+            -->
+            <t t-foreach="tax_totals['groups_by_total_type'].items()" t-as="total_type_groups">
+                <t t-if="total_type_groups[0] in total_types_to_show" t-foreach="total_type_groups[1]" t-as="amount_by_group">
+                    <tr>
+                        <t t-if="multiple_tax_groups or tax_totals['amount_untaxed'] != amount_by_group['tax_group_base_amount']">
```

Comment:
`tax_totals['amount_untaxed'] != amount_by_group['tax_group_base_amount']` could be a bad idea if you are dealing with zero taxes. What do you think?